frappe/public/js/frappe/form/grid.js

```diff
@@ -446,7 +449,7 @@ export default class Grid {
 				grid_row.doc = d;
 				grid_row.refresh();
 			} else {
-				var grid_row = new GridRow({
+				var grid_row = new this.GridRow({
```

Comment:
@dj12djdjs I can merge this for now, but customizations like these are very hard to keep supporting over time and will probably get lost in some refactors without tests very easily. Overriding in general creates problems because two apps can't override same things. 

What I'd recommend based on use cases you mentioned is just exposing the existing `GridRow` via some events where you can hook into it and modify behaviour. 


Many examples of this in our code. e.g. `{child_table}_add` is an event triggered when adding new row. Similarly, you can add events for the lifecycle of Grid.

Reply:
> What I'd recommend based on use cases you mentioned is just exposing the existing `GridRow` via some events where you can hook into it and modify behaviour.
> 
> Many examples of this in our code. e.g. `{child_table}_add` is an event triggered when adding new row. Similarly, you can add events for the lifecycle of Grid.

@ankush I agree, this is a much better implementation. I will think about how to add life cycle hooks to grid.